libs/SmartSync/SmartSyncTests/SFLayoutSyncManagerTests.m

```diff
@@ -0,0 +1,174 @@
+    [self validateResult:objType layout:layoutData];
+}
+
+/**
+ * Test for fetching layout in SFSDKFetchModeCacheFirst mode with an empty cache.
+ */
+- (void)testFetchLayoutInCacheFirstModeWithoutCacheData {
+    __block NSString *objType = nil;
+    __block SFLayout *layoutData = nil;
+    XCTestExpectation *fetchLayoutCacheFirst = [self expectationWithDescription:@"fetchLayoutCacheFirst"];
+    [self.layoutSyncManager fetchLayoutForObject:kAccount layoutType:kCompact mode:SFSDKFetchModeCacheFirst completionBlock:^(NSString *objectType, SFLayout *layout) {
+        objType = objectType;
+        layoutData = layout;
+        [fetchLayoutCacheFirst fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
+    [self validateResult:objType layout:layoutData];
+}
+
+/**
+ * Test for fetching layout in SFSDKFetchModeServerFirst mode.
+ */
+- (void)testFetchLayoutInServerFirstMode {
+    __block NSString *objType = nil;
+    __block SFLayout *layoutData = nil;
+    XCTestExpectation *fetchLayoutServerFirst = [self expectationWithDescription:@"fetchLayoutServerFirst"];
+    [self.layoutSyncManager fetchLayoutForObject:kAccount layoutType:kCompact mode:SFSDKFetchModeServerFirst completionBlock:^(NSString *objectType, SFLayout *layout) {
+        objType = objectType;
+        layoutData = layout;
+        [fetchLayoutServerFirst fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
+    [self validateResult:objType layout:layoutData];
+}
+
+/**
+ * Test for fetching layout multiple times and ensuring only 1 row is created.
+ */
+- (void)testFetchLayoutMultipleTimes {
+    __block NSString *objType = nil;
+    __block SFLayout *layoutData = nil;
+    XCTestExpectation *fetchLayoutOne = [self expectationWithDescription:@"fetchLayoutOne"];
+    [self.layoutSyncManager fetchLayoutForObject:kAccount layoutType:kCompact mode:SFSDKFetchModeServerFirst completionBlock:^(NSString *objectType, SFLayout *layout) {
+        objType = objectType;
+        layoutData = layout;
+        [fetchLayoutOne fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
+    [self validateResult:objType layout:layoutData];
+    XCTestExpectation *fetchLayoutTwo = [self expectationWithDescription:@"fetchLayoutTwo"];
+    [self.layoutSyncManager fetchLayoutForObject:kAccount layoutType:kCompact mode:SFSDKFetchModeServerFirst completionBlock:^(NSString *objectType, SFLayout *layout) {
+        objType = objectType;
+        layoutData = layout;
+        [fetchLayoutTwo fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
+    [self validateResult:objType layout:layoutData];
+    SFQuerySpec *querySpec = [SFQuerySpec newSmartQuerySpec:[NSString stringWithFormat:kQuery, kSoupName, kSoupName, kSoupName, kAccount, kCompact] withPageSize:1];
+    long numRows = [self.layoutSyncManager.smartStore countWithQuerySpec:querySpec error:nil];
```

Comment:
It can’t be more than 1 since the pageSize is 1.
Should use 2 or do a select count